src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,162 @@
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong("id"));
+            }
+            addRolesToDB(user, connection);
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creation user " + user + " failed", e);
+        }
+    }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM users u "
+                    + "JOIN users_roles ur ON u.id = ur.user_id "
+                    + "JOIN roles r on ur.role_id = r.id "
+                    + "WHERE u.id = ? AND deleted = FALSE");
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getUserFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting user with id " + id + " failed", e);
+        }
+    }
+
+    @Override
+    public Optional<User> getByLogin(String login) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM users u "
+                    + "JOIN users_roles ur ON u.id = ur.user_id "
+                    + "JOIN roles r on ur.role_id = r.id "
+                    + "WHERE u.login = ? AND deleted = FALSE");
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getUserFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting user with login " + login + " failed", e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM users u "
+                    + "JOIN users_roles ur ON u.id = ur.user_id "
+                    + "JOIN roles r on ur.role_id = r.id "
+                    + "WHERE deleted = FALSE ORDER BY u.id",
+                    ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
+            ResultSet resultSet = statement.executeQuery();
+            List<User> allUsers = new ArrayList<>();
+            while (resultSet.next()) {
+                allUsers.add(getUserFromResultSet(resultSet));
+                resultSet.previous();
```

Comment:
so you stay at the same point all the time?

Reply:
I am doing step back after the last calling next() at line 159